### ISSUE TITLE
DynamicComponentLoader not initializing properly in some circumstances

### ISSUE THREAD

**Issue by icnec** [angular/angular#7044]
I have a tooltip directive that takes a dynamiccomponentloader in the constructor.  The loader comes into the constructor fully initialized when the directive is found within existing components on the page.  However, if a component containing tooltip directives is dynamically loaded (via the dynamiccomponentloader), the directive is still detected, but the loader within the constructor of the directive is stripped of all of the loader functionality.  The _compiler and _viewmanager properties are no longer present, and none of the dynamic loader methods exist to be called.

**Comment by wendro** [comment]
I've observed problem with working with DynamicComponentLoader with BETA-6 that it under some circumstances does CONSTRUCT components but it does not execute their LIFE CYCLE. They are created & shown but angular2 life cycle methods such as ngOnInit, ngAfterContentInit are not triggered at all. If I will find a time will try to drill down on this a bit. 

In short what I'm doing is I've created DynamicContainerComponent which has @Input('items') and @Input('load-strategy') which is rendering controls through dynamically passed configuration using DynamicComponentLoader.
